### PR TITLE
More weapon tracking

### PR DESCRIPTION
**Pull request by nicoeg**
There we some cases where a player suddenly had a lot of weapons and there still is some but cannot currently recreate.

Added the flashbang event to track when it drops only to find out that a dropped event is also logged.. Didn't want to remove it now that I had added it. Let me know if you don't like the `weapon.ex` refactor

### CODE REVIEW

File: lib/csgo_stats/logs/parser/weapon.ex [hunk 1]
```diff
@@ -1,6 +1,54 @@
...
+    string("usp_silencer"),
+    string("xm1014")
+  ]
+
+  @grenades [
+    string("smokegrenade"),
+    string("flashbang"),
+    string("incgrenade"),
+    string("hegrenade"),
+    string("inferno")
```
Comment (trarbr) [inline lib/csgo_stats/logs/parser/weapon.ex]:
```suggestion
    string("inferno"),
    string("molotov")
```

File: lib/csgo_stats/logs/parser/weapon.ex [hunk 2]
```diff
@@ -1,6 +1,54 @@
...
+    string("glock"),
+    string("hkp2000"),
+    string("knife_t"),
+    string("knife"),
+    string("m249"),
+    string("m4a1_silencer"),
+    string("m4a1"),
+    string("mac10"),
+    string("mag7"),
+    string("molotov"),
```
Comment (trarbr) [inline lib/csgo_stats/logs/parser/weapon.ex]:
```suggestion
```

File: lib/csgo_stats/matches/match.ex [hunk 3]
```diff
@@ -33,6 +33,7 @@ defmodule CsgoStats.Matches.Match do
         :defuser -> %{player | defuser: true}
         :vesthelm -> %{player | armor: 100, helmet: true}
         :vest -> %{player | armor: 100}
+        :knife -> %{player | weapons: player.weapons}
```
Comment (trarbr) [inline lib/csgo_stats/matches/match.ex]:
```suggestion
        :knife -> player
        :knife_t -> player
```

**Review by trarbr — COMMENTED** [review]
I love it! I wonder if we'll have to do some sorting of weapons - e.g. I have a pistol and rifle, I kill someone and drop my pistol to pick up their pistol - now it'll show rifle first and pistol second, right? Same with nades...  But it's fine with me if we handle this later.

**Comment by nicoeg** [comment]
>  I wonder if we'll have to do some sorting of weapons - e.g. I have a pistol and rifle, I kill someone and drop my pistol to pick up their pistol - now it'll show rifle first and pistol second, right? Same with nades...

Yeah we'll probably have to do that 🤔

File: lib/csgo_stats/logs/parser/weapon.ex [hunk 4]
```diff
@@ -1,6 +1,54 @@
...
+    string("xm1014")
+  ]
+
+  @grenades [
+    string("smokegrenade"),
+    string("flashbang"),
+    string("incgrenade"),
+    string("hegrenade"),
+    string("inferno"),
+    string("molotov")
```
Comment (trarbr) [inline lib/csgo_stats/logs/parser/weapon.ex]:
```suggestion
    string("molotov"),
    string("decoy")
```

File: lib/csgo_stats/logs/parser/weapon.ex [hunk 5]
```diff
@@ -1,6 +1,54 @@
...
 
+  @weapons [
+    string("ak47"),
+    string("aug"),
+    string("awp"),
+    string("bizon"),
+    string("c4"),
+    string("cz75a"),
+    string("deagle"),
+    string("decoy"),
```
Comment (trarbr) [inline lib/csgo_stats/logs/parser/weapon.ex]:
```suggestion
```

**Review by trarbr — CHANGES_REQUESTED** [review]
Noticed a grenade more. When that is fixed, this is good to merge

File: lib/csgo_stats/logs/parser/weapon.ex (hunk 4 above)
Comment (nicoeg) [reply lib/csgo_stats/logs/parser/weapon.ex]: Whoops :D

**Review by trarbr — APPROVED** [review]

**Pull request merged**